### DISCUSSION TITLE
Is there any way to show all symbols in outline (modal and panel)?

### DISCUSSION THREAD

**Discussion by pavelpuchok** [zed-industries/zed#19278] [Feature Requests]
I can't see all possible symbols in Outline panel/modal for JS objects. For example object fields are missing.

To be more specific let's look for example. I have the following JS file:
```javascript
const someConst = {
  nested: {
    foo: {
      fooParams: {
        fooParam: 'foo'
      },
      fooHandler: () => {
        return '';
      },
    },
    bar: {
      barParams: {
        booParam: 'bar'
      },
      barHandler() {
        return '';
      },
    },
  }
};
```
And thats what Outline panel looks like:
![image](https://github.com/user-attachments/assets/88d6197b-cc2d-420b-8253-6fd559343ec3)

But I want to see `nested`, `foo`, `fooParams` and all other fields as well. I don't have an goland installed anymore, but I pretty sure that theirs Structure layout showed up everything.

Is it possible to configure somehow?

**Comment by SomeoneToIgnore** [comment]
I do not think there's a simple way to configure this, and the absence of other definitions means tree-sitter queries in https://github.com/zed-industries/zed/blob/4fe40a8064994aeb0a1a9f0e1bc72e83c972b01a/crates/languages/src/go/outline.scm#L1 are not complete/incorrect.

**Reply by pavelpuchok** [reply]
Thank you for response. Is these queries files is something that I can change on the fly or compilation is needed?

**Reply by SomeoneToIgnore** [reply]
There's no good support for that, but for some extensions it's possible to do this until they update/redownload.
Extensions store their *.scm files locally, e.g. for macOS ` find ~/Library/Application\ Support/Zed -type f -name "*.scm"` but there's no  `go` file as `go` extension is bundled with Zed.

In this case, IIRC we bundle the files within Zed binary itself, hence it's not possible to change without opening a PR.
But I might be wrong about that binary bundling part.

**Comment by JamieMason** [comment]
Hi all, I stumbled on this discussion when looking for the same thing. I'm hoping to be able to see the same kind of information as the panel on the far left but only see what seems to be top level symbols and its public methods only.

Is the status of this ~feature request~ **discussion about a desired feature** that this is something which isn't surfaced by tree-sitter? Did I understand that correctly?

<img src="https://github.com/user-attachments/assets/71911ce9-93ef-493e-8750-ad756ac0a0e2" width=300>

Thanks a lot.

**Reply by SomeoneToIgnore** [reply]
> this feature request

This is a discussion, not a feature request.

> isn't surfaced by tree-sitter

Not surfaced by the Zed's tree-sitter queries in the corresponding language's outline.scm
https://github.com/zed-industries/zed/blob/main/crates/languages/src/typescript/outline.scm

**Reply by JamieMason** [reply]
Thanks @SomeoneToIgnore 👍

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
